Update eloston-chromium from 97.0.4692.71 to 97.0.4692.99

frodera opened the pull request:
Also generalise the use of epoch time for arm64 and x86-64
as it now used for releases of both achitectures in
https://github.com/kramred/ungoogled-chromium-macos/releases/

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask <cask>` is error-free.
- [ ] `brew style --fix <cask>` reports no offenses.

frodera commented:
`livecheck` works.

```shell
% brew livecheck --debug Casks/eloston-chromium.rb

Cask:             eloston-chromium
Livecheckable?:   Yes

URL:              https://github.com/kramred/ungoogled-chromium-macos/releases/
Strategy:         PageMatch

Matched Versions:
97.0.4692.99-1.1,x86-64,1642829547
eloston-chromium : 97.0.4692.99-1.1,x86-64,1642829547 ==> 97.0.4692.99-1.1,x86-64,1642829547
```

neutric commented on Casks/eloston-chromium.rb:
```diff
@@ -1,32 +1,28 @@
 
+  livecheck do
+    url "https://github.com/kramred/ungoogled-chromium-macos/releases/"
+    strategy :page_match do |page|
```
The [docs](https://docs.brew.sh/Brew-Livecheck#url-guidelines) on `livecheck` say
> [...] avoid checking the release page for a GitHub project because the latest stable version can be pushed off the first page by pre-release versions. In this scenario, it’s more reliable to use the Git strategy, which fetches all the tags in the repository.

See e.g. [`duplicati.rb`](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/duplicati.rb#L13-L20) for a good example.

frodera replied:
That's a very valid point, but, if I'm reading the docs correctly, for this particular scenario it may not be the best approach. `ungoogled-chromium-macos` builds take a very long time. From the moment the tag is created until the artifacts are released and ready to download several hours usually pass. 

For the most recent release it took [16 hours](https://github.com/kramred/ungoogled-chromium-macos/actions/runs/1728985705)  for the x86-64 architecture and [almost 27 hours](https://github.com/kramred/ungoogled-chromium-macos/actions/runs/1728985180) for arm64.

IMO, it would be confusing that `livecheck` reported a new version is available, based on the Git tag, but it would still take almost a day for the associated artifact to be ready for download and install.

neutric commented on Casks/eloston-chromium.rb:
```diff
@@ -1,32 +1,28 @@
+  livecheck do
+    url "https://github.com/kramred/ungoogled-chromium-macos/releases/"
+    strategy :page_match do |page|
+      match = page.match(%r{releases/download/(.+)[._-]#{version.csv.second}[._-][._-](\d+)/ungoogled[._-]chromium[._-](.+)[._-]macos\.dmg}i)
```
Versions like `97.0.4692.99` are matched with `(\d+(?:\.\d+)+)` instead of `(.+)`, because the latter is generally too insensitive to format changes. To account for the `-` in the slightly more complicated `97.0.4692.99-1.1`, use `(\d+(?:[.-]\d+)+)`.

`[._-][._-]` can be written as `[._-]{2}`

frodera replied:
Makes sense 👍🏻 . Fixed.

neutric commented on Casks/eloston-chromium.rb:
```diff
@@ -1,32 +1,28 @@
-
-    url "https://github.com/kramred/ungoogled-chromium-macos/releases/download/#{version}/ungoogled-chromium_#{version}-macos.dmg",
-        verified: "github.com/kramred/ungoogled-chromium-macos/"
+    version "97.0.4692.99-1.1,x86-64,1642829547"
```
Instead of having `x86-64` and `arm64` as part of the `version`, define the variable `arch` and use it accordingly as seen in [`corretto.rb`](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/corretto.rb#L2).

frodera replied:
Good point. Fixed.

neutric reviewed (commented):
Thanks for your contribution to `homebrew-cask`, @frodera! My review is in the code. Take it with a grain of salt as I am not a maintainer.

frodera commented:
> Thanks for your contribution to `homebrew-cask`, @frodera! My review is in the code. Take it with a grain of salt as I am not a maintainer.

Thanks a lot @neutric for your feedback. I have addressed your comments and pushed a new commit.

bevanjkay commented on Casks/eloston-chromium.rb:
```diff
@@ -1,32 +1,32 @@
-        "#{match[1]},#{match[2]}"
-      end
-    end
+    version "97.0.4692.99-1.1,#{arch},1642868666"
```
@frodera Is there a reason why the `#{arch}` string is being included in the versions?

miccal replied:
Ah, I see what you are driving at now -- did not think this through carefully enough.

miccal commented:
For Intel:
```
|-> brew livecheck eloston-chromium --debug

Cask:             eloston-chromium
Livecheckable?:   Yes

URL:              https://github.com/kramred/ungoogled-chromium-macos/releases/
Strategy:         PageMatch

Matched Versions:
97.0.4692.99-1.1,x86-64,1642829547
eloston-chromium : 97.0.4692.99-1.1,x86-64,1642829547 ==> 97.0.4692.99-1.1,x86-64,1642829547
```
```
|-> brew fetch eloston-chromium
==> Downloading https://github.com/kramred/ungoogled-chromium-macos/releases/download/97.0.4692.99-1.1_x86-64__1642829547/ungoogled-chromium_97.0.4692.99-1.1_x86-64-macos.dmg
==> Downloading from https://objects.githubusercontent.com/github-production-release-asset-2e65be/192515338/73740440-2103-489e-aaab-f8390eeaeee0?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20220123%2Fus-east
######################################################################## 100.0%
Downloaded to: /Users/miccal/Library/Caches/Homebrew/downloads/7ad866712e7eb886cd7463ad93b787441502e2de2647ae5382cd0f837e1a99dd--ungoogled-chromium_97.0.4692.99-1.1_x86-64-macos.dmg
SHA256: cb62374873c04497b2e257cec8fa81565d8147311ddfda13f552acac8ec747f3
```
For Arm:
```
|-> brew livecheck eloston-chromium --debug

Cask:             eloston-chromium
Livecheckable?:   Yes

URL:              https://github.com/kramred/ungoogled-chromium-macos/releases/
Strategy:         PageMatch

Matched Versions:
97.0.4692.99-1.1,arm64,1642868666
eloston-chromium : 97.0.4692.99-1.1,arm64,1642868666 ==> 97.0.4692.99-1.1,arm64,1642868666
```
```
|-> brew fetch eloston-chromium
==> Downloading https://github.com/kramred/ungoogled-chromium-macos/releases/download/97.0.4692.99-1.1_arm64__1642868666/ungoogled-chromium_97.0.4692.99-1.1_arm64-macos.dmg
==> Downloading from https://objects.githubusercontent.com/github-production-release-asset-2e65be/192515338/ab37e086-1cb8-4336-ae21-136ed0087142?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20220124%2Fus-east
######################################################################## 100.0%
Downloaded to: /Users/miccal/Library/Caches/Homebrew/downloads/6f9a27919d4315224c67a9b856a5a33fcdd317516a056e132e789a0b70b77a59--ungoogled-chromium_97.0.4692.99-1.1_arm64-macos.dmg
SHA256: 790164b7b166c22fad1e830523e6eb3abc41b54b57ce8c7c8ca2a5dfbe5ca282
```

miccal commented:
Great work @frodera, thank you.
